src/menu/menu.c

```diff
@@ -265,34 +276,66 @@ separator_create(struct menu *menu, const char *label)
 	struct menuitem *menuitem = znew(*menuitem);
 	menuitem->parent = menu;
 	menuitem->selectable = false;
+	menuitem->type = label ? LAB_MENU_TITLE : LAB_MENU_SEPARATOR_LINE;
 	struct server *server = menu->server;
 	struct theme *theme = server->theme;
-	menuitem->height = theme->menu_separator_line_thickness +
-			2 * theme->menu_separator_padding_height;
```

Comment:
menuitem->height needs check for being zero, otherwise you get funny results
copy lines 197-200 and you should be fine.